src/nxt_socket_msg.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) Igor Sysoev
+ * Copyright (C) NGINX, Inc.
+ */
+
+#include <nxt_main.h>
+#include <nxt_socket_msg.h>
+
+
+#if (NXT_HAVE_MSGHDR_MSG_CONTROL)
+
+
+ssize_t
+nxt_sendmsg(nxt_socket_t s, nxt_iobuf_t *iob, nxt_uint_t niob, 
+    const void *oob, size_t oobn)
+{
+    struct msghdr   msg;
+    
+    msg.msg_name = NULL;
+    msg.msg_namelen = 0;
+    msg.msg_iov = iob;
+    msg.msg_iovlen = niob;
+    /* Flags are cleared just to suppress valgrind warning. */
+    msg.msg_flags = 0;
+    msg.msg_control = (void *) oob;
+    msg.msg_controllen = oobn;
+
+    if (oobn == 0) {
+        msg.msg_control = NULL;
+    }
+
+    return sendmsg(s, &msg, 0);
+}
+
+ssize_t
+nxt_recvmsg(nxt_socket_t s, nxt_iobuf_t *iob, nxt_uint_t niob,
+    void *oob, size_t *oobn)
+{
+    ssize_t                n;
+    struct msghdr          msg;
+    
+    msg.msg_name = NULL;
+    msg.msg_namelen = 0;
+    msg.msg_iov = iob;
+    msg.msg_iovlen = niob;
+    msg.msg_control = oob;
+    msg.msg_controllen = *oobn;
+
+    n = recvmsg(s, &msg, 0);
+
+    if (nxt_fast_path(n != -1)) {
+        *oobn = msg.msg_controllen;
+    }
+
+    return n;
+}
+
+
+void
+nxt_socket_msg_set_oob(u_char *oob, size_t *oobn, int fd)
+{
+    struct msghdr  msg;
+    struct cmsghdr *cmsg;
+
+    msg.msg_control    = (void *) oob;
+    msg.msg_controllen = *oobn;
+    
+    *oobn = 0;
+    cmsg = (struct cmsghdr *) oob;
+
+#if (NXT_HAVE_MSGHDR_CMSGCRED)
+
+    cmsg->cmsg_len = CMSG_LEN(sizeof(struct NXT_CRED_STRUCT));
+    cmsg->cmsg_level = SOL_SOCKET;
+    cmsg->cmsg_type = NXT_CRED_CMSGTYPE;
+
+    *oobn += CMSG_SPACE(sizeof(struct NXT_CRED_STRUCT));
+    cmsg = CMSG_NXTHDR(&msg, cmsg);
+#endif
+
+    if (fd != -1) {
+
```

Comment:
Blank line is not required here.

Comment need to have aligned '*'.

Remove trailing spaces.